src/Docfx.Dotnet/DotnetApiCatalog.Toc.cs

```diff
@@ -120,9 +122,12 @@ IEnumerable<TocNode> CreateNamespaceToc(INamespaceSymbol ns)
                 }
 
                 node.containsLeafNodes = node.items.Any(i => i.containsLeafNodes);
-                if (!idExists && node.containsLeafNodes)
+                if (node.containsLeafNodes)
```

Comment:
It seems that the root cause is we assume in line 96 if a TOC node exists in `tocNodes` it'll be included in the final. Does the following code fixes the problem?

```suggestion
                if (!node.containsLeafNodes)
                {
                    tocNodes.Remove(node);
                }

                if (!idExists && node.containsLeafNodes)
                {
                    yield return node;
                }
```